### ISSUE TITLE
IPython 8.2.0 ERROR with %cpaste -q

### ISSUE THREAD

**Issue by ljmc-github** [ipython/ipython#13622]
I recently updated my workflow to use pyenv, which brought along an update to all my pip installed tools.

IPython was updated from 8.1.1 to 8.2.0.

Since then running a block with `%cpaste -q` which I use a lot causes an ERROR:

```
ERROR! Session/line number was not unique in database. History logging moved to new session 3
```

I compared with my old install and with a downgraded version, both work as expected: 

1. Old install using brew and pip
```
Python 3.10.2 (main, Feb  2 2022, 06:19:27) [Clang 13.0.0 (clang-1300.0.29.3)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.1.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %cpaste -q
i = 1
print(i)
--
1

In [2]:
```

2. New install in pyenv and pip
```
Python 3.9.11 (main, Apr  4 2022, 12:07:49)
Type 'copyright', 'credits' or 'license' for more information
IPython 8.2.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %cpaste -q
j = 2
print(j)
--
2

ERROR! Session/line number was not unique in database. History logging moved to new session 3
In [3]:
```

3. Downgraded to 8.1.1
```
Python 3.9.11 (main, Apr  4 2022, 12:07:49)
Type 'copyright', 'credits' or 'license' for more information
IPython 8.1.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %cpaste -q
i = 1
print(i)
--
1

In [2]:
```
 
There was a difference in the history.sqlite database, both 8.1.1 version has a single row for the `%cpaste -q`, the 8.2.0 was also showing the `j = 2` and `print(j)` on a second row, but I'm unable to find this output at this time. The overincrementing can be seen in the jump from prompt 1 straight to 3.

**Comment by Carreau** [comment]
Yeah, I know what this is, there was a recent modification so that the content of `%paste` instead of `%paste` itself would be put in history. I probably forgot about cpaste.

**Comment by Paul-Aime** [comment]
I don't know if that helps, but it also happens with `%paste`.

- `%paste` : overincrementing
- `%paste -q` : overincrementing + history database error
- `%paste -r`: no issue

```python
In [1]: %paste
print('hi')

## -- End pasted text --
hi

In [3]: %paste -q
hi

ERROR! Session/line number was not unique in database. History logging moved to new session 2
In [5]: %paste -r
Re-executing 'print('hi')...' (11 chars)
hi

In [6]: 
```

**Comment by euglevi** [comment]
I got the same with iPython 8.3.0. Is the only solution to go back to version  8.1.1?

**Comment by jdtsmith** [comment]
Same here.  Since a tool I'm developing uses `cpaste -q` extensively to send blocks of code to IPython, this is especially problematic.

**Comment by argapost** [comment]
Same in iPython 8.5.0 has anyone found a workaround?

**Comment by CosmosAtlas** [comment]
I've been bothered by this soooo much, I've took a look at the code and attempted a fix.

Let's wait and see how the ipython maintainers think.

**Comment by ljmc-github** [comment]
I just tested [8.6.0](https://github.com/ipython/ipython/releases/tag/8.6.0), which includes @CosmosAtlas's fixes #13763, looks fixed, closing for now.
